### ISSUE TITLE
`ttnn.reduce_scatter` with ring topology has inconsistent rounding error

### ISSUE THREAD

**Issue by rdraskicTT** [tenstorrent/tt-metal#35087]
`ttnn.reduce_scatter` and `ttnn.experimental.reduce_scatter_minimal_async` are creating differing outputs per user in batched prefill for llama 70B. We need user logits to be the same for same inputs in order to have deterministic execution.

Unit test repro can be found at `rdraskic/reduce-scatter-bug` and can be run with `pytest test_reduce_scatter.py`. Changing the topology to linear or precision to bfloat16 makes the test pass but is unacceptable on the model side since it affects performance.

**Comment by rdraskicTT** [comment]
Reshape before reduce scatter `distributed_tensor = ttnn.reshape(distributed_tensor, (1, 32, 128, 1280)` makes the test pass.

**Comment by rdraskicTT** [comment]
Running test with watcher gives the following assertion
```
2025-12-29 11:32:18.619 | critical |          Always | TT_THROW: Failed to generate binaries for ring_reduce_scatter_minimal_async_writer TT_THROW @ /localdev/rdraskic/tt-metal/tt_metal/jit_build/build.cpp:55: tt::exception
info:
brisc build failed. Log: In file included from /localdev/rdraskic/tt-metal/tt_metal/hw/inc/tt-1xx/wormhole/noc_nonblocking_api.h:13,
                 from /localdev/rdraskic/tt-metal/tt_metal/hw/inc/tt-1xx/risc_common.h:192,
                 from /localdev/rdraskic/tt-metal/tt_metal/hw/firmware/src/tt-1xx/brisck.cc:8:
/localdev/rdraskic/tt-metal/tt_metal/fabric/hw/inc/api_common.h: In function 'void tt::tt_fabric::common::experimental::populate_unicast_scatter_write_fields(volatile tt::tt_fabric::LowLatencyPacketHeader*, uint16_t, const CommandHeaderT&)':
/localdev/rdraskic/tt-metal/tt_metal/fabric/hw/inc/api_common.h:234:17: error: cannot convert 'const char*' to 'debug_assert_type_t' [-Wtemplate-body]
  234 |                 "scatter chunk_count must be between 2 and 4");
```

However, this is something that gets triggered even when test is passing by setting different parameters.
Once I removed this assertion and rerun the test nothing else got reported by the watcher.

**Comment by jvegaTT** [comment]
In the above I assume that you mean you are reshaping from 1,1,4096,128 to 1,32,128,128 before performing the reduction operation. In that case this is a 0 cost view which means it does not affect perf at all to keep the workaround in.

I have verified locally and the view function which is guaranteed to be 0 cost fixes the issue completely

<img width="1494" height="834" alt="Image" src="https://github.com/user-attachments/assets/06ccc965-0c7a-4bb6-9e46-5542d0dd5c12" />

What this means is the workaround does not affect perf at all, I will still investigate a fix but you should be completely unblocked.

**Comment by jvegaTT** [comment]
Lowering the priority as it no longer blocks or impacts perf

**Comment by jvegaTT** [comment]
Replacing with this issue as it is no longer related to the model perf https://github.com/tenstorrent/tt-metal/issues/35319
